### ISSUE TITLE
pulldown_cmark_wikilink non-exhaustive patterns

### ISSUE THREAD

**Issue by Peyton-Spencer** [rambip/leptos-markdown#9]
I believe pulldown_cmark was updated under you and some types were added; I attached the compiler output below.

Here is my branch which simply adds 3 `todo!()` for the missing match arms. I could help write the render fn.
https://github.com/Peyton-Spencer/leptos-markdown
<img width="637" alt="image" src="https://github.com/rambip/leptos-markdown/assets/6766068/b82a758e-198a-4203-892b-7b9f712f0f02">

```
error[E0004]: non-exhaustive patterns: `pulldown_cmark_wikilink::Event::InlineHtml(_)` not covered
   --> /Users/peyton/.cargo/git/checkouts/leptos-markdown-8bc941d757f86426/fb2cc7c/src/render.rs:137:30
    |
137 |         let rendered = match item {
    |                              ^^^^ pattern `pulldown_cmark_wikilink::Event::InlineHtml(_)` not covered
    |
note: `pulldown_cmark_wikilink::Event<'_>` defined here
   --> /Users/peyton/.cargo/git/checkouts/pulldown-cmark-f7479fa051ef81e1/f923db2/src/lib.rs:341:5
    |
322 | pub enum Event<'a> {
    | ------------------
...
341 |     InlineHtml(CowStr<'a>),
    |     ^^^^^^^^^^ not covered
    = note: the matched value is of type `pulldown_cmark_wikilink::Event<'_>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
156 ~             Math(disp, content) => render_maths(self.context, &content, &disp, range),
157 ~             pulldown_cmark_wikilink::Event::InlineHtml(_) => todo!(),
    |

error[E0004]: non-exhaustive patterns: `pulldown_cmark_wikilink::Tag::HtmlBlock` not covered
   --> /Users/peyton/.cargo/git/checkouts/leptos-markdown-8bc941d757f86426/fb2cc7c/src/render.rs:281:18
    |
281 |         Ok(match tag.clone() {
    |                  ^^^^^^^^^^^ pattern `pulldown_cmark_wikilink::Tag::HtmlBlock` not covered
    |
note: `pulldown_cmark_wikilink::Tag<'_>` defined here
   --> /Users/peyton/.cargo/git/checkouts/pulldown-cmark-f7479fa051ef81e1/f923db2/src/lib.rs:156:5
    |
135 | pub enum Tag<'a> {
    | ----------------
...
156 |     HtmlBlock,
    |     ^^^^^^^^^ not covered
    = note: the matched value is of type `pulldown_cmark_wikilink::Tag<'_>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
350 ~             },
351 +             pulldown_cmark_wikilink::Tag::HtmlBlock => todo!()
    |

error[E0004]: non-exhaustive patterns: `&pulldown_cmark_wikilink::Tag::HtmlBlock` not covered
   --> /Users/peyton/.cargo/git/checkouts/leptos-markdown-8bc941d757f86426/fb2cc7c/src/utils.rs:4:11
    |
4   |     match t {
    |           ^ pattern `&pulldown_cmark_wikilink::Tag::HtmlBlock` not covered
    |
note: `pulldown_cmark_wikilink::Tag<'_>` defined here
   --> /Users/peyton/.cargo/git/checkouts/pulldown-cmark-f7479fa051ef81e1/f923db2/src/lib.rs:156:5
    |
135 | pub enum Tag<'a> {
    | ----------------
...
156 |     HtmlBlock,
    |     ^^^^^^^^^ not covered
    = note: the matched value is of type `&pulldown_cmark_wikilink::Tag<'_>`
help: ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown
    |
21  ~         Tag::MetadataBlock(k) => TagEnd::MetadataBlock(*k),
22  ~         &pulldown_cmark_wikilink::Tag::HtmlBlock => todo!(),
    |
    ```

**Comment by rambip** [comment]
Okay, I will try to fix that.
I know that pulldown-cmark is undergoing some big changes, so I will have to fix other things in the future

**Comment by rambip** [comment]
Closed by 1a407f4

**Comment by Peyton-Spencer** [comment]
Thank you!
